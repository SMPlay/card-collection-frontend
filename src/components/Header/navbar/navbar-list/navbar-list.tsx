import React, { useState } from "react";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import { Link } from "react-router-dom";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import ExpandLess from "@material-ui/icons/ExpandLess";
import ExpandMore from "@material-ui/icons/ExpandMore";
import { Collapse } from "@material-ui/core";
import { ApolloError } from "@apollo/client";

import { PageType } from "../../../../types/PageType";
import { CollectionNameType } from "../../../../types/CollectionNameType";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    navlink: { color: "rgb(79, 79, 79)" },
    nested: { paddingLeft: theme.spacing(4) },
  })
);

export interface NavbarListProps {
  pages: PageType[];
  collectionsName: CollectionNameType[] | undefined;
  collectionsNameError: boolean | ApolloError | undefined;
}

export const NavbarList: React.FC<NavbarListProps> = ({
  pages,
  collectionsName,
  collectionsNameError
}) => {
  const [isOpen, setOpen] = useState<boolean>(false);

  const classes = useStyles();

  const handleClick = () => {
    setOpen(!isOpen);
  };

  return (
    <List>
      {pages.map((page) => (
        <Link to={page.url} key={page.pageName}>
          <ListItem button>
            <ListItemText
              classes={{ primary: classes.navlink }}
              primary={page.pageName}
            />
          </ListItem>
        </Link>
      ))}
      <ListItem button onClick={handleClick}>
        <ListItemText primary="Коллекции" />
        {isOpen ? <ExpandLess /> : <ExpandMore />}
      </ListItem>
      <Collapse in={isOpen} timeout="auto" unmountOnExit>
        <List component="div" disablePadding>
          {collectionsNameError ? (
            <ListItem button className={classes.nested}>
              <ListItemText primary="Проблемы с сервером" data-test-error="error-message"/>
            </ListItem>
          ) : (
            collectionsName?.map((collection) => (
              <Link to={`collections/${collection.id}`} key={collection.id}>
                <ListItem button className={classes.nested}>
                  <ListItemText classes={{ primary: classes.navlink }} primary={collection.name} />
                </ListItem>
              </Link>
            ))
          )}
        </List>
      </Collapse>
    </List>
  );
};
