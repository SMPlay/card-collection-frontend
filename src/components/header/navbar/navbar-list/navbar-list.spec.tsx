import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

import { NavbarList, NavbarListProps } from "./navbar-list";

describe("Navbar list component", () => {
  let props: NavbarListProps;

  beforeEach(() => {
    props = {
      pages: [
        { pageName: "Test", url: "/test"}
      ],
      collectionsName: [
        { id: "test", name: "test" }
      ],
      collectionsNameError: false
    };
  });

  it("should render", () => {
    const tree = renderer.create(
      <Router>
        <NavbarList {...props}/>
      </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});