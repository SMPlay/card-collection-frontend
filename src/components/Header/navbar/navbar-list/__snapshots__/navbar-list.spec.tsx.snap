// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar list component should render 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <a
    href="/test"
    onClick={[Function]}
  >
    <div
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary makeStyles-navlink-1 MuiTypography-body1 MuiTypography-displayBlock"
        >
          Test
        </span>
      </div>
    </div>
  </a>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        Коллекции
      </span>
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
      />
    </svg>
  </div>
</ul>
`;
